File: src/test/java/net/codinux/csv/kcsv/reader/RowHandlerTest.java
package net.codinux.csv.kcsv.reader

import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test

class RowHandlerTest {
  @Test
  fun test() {
    val rh = RowHandler(1)
    rh.add("foo")
    rh.add("bar")
    val csvRow = rh.buildAndReset()
    Assertions.assertNotNull(csvRow)
    Assertions.assertEquals(
      "CsvRow[originalLineNumber=1, fields=[foo, bar], comment=false]",
      csvRow.toString()
    )
  }
}

File: src/commonTest/kotlin/net/codinux/csv/reader/RowHandlerTest.kt
package net.codinux.csv.reader

import kotlin.test.Test
import kotlin.test.assertEquals
import kotlin.test.assertNotNull

class RowHandlerTest {
  @Test
  fun test() {
    val rh = RowHandler(1, false, emptySet(), false)
    val buffer ="foo,bar".toCharArray()

    rh.add(buffer, 0, 3, 0, '"')
    rh.add(buffer, 4, 7, 0, '"')

    val csvRow = rh.buildAndReset()
    assertNotNull(csvRow)
    assertEquals(
      "CsvRow[originalLineNumber=1, fields=[foo, bar], comment=false]",
      csvRow.toString()
    )
  }
}